cmd/collector/app/builder/span_handler_builder_test.go

```diff
@@ -95,6 +97,35 @@ func TestNewSpanHandlerBuilderMemorySet(t *testing.T) {
 	assert.NotNil(t, zHandler)
 }
 
+func TestNewSpanHandlerBuilderElasticSearch(t *testing.T) {
+	originalArgs := os.Args
+	defer func() {
+		os.Args = originalArgs
+	}()
+	os.Args = []string{"test", "--span-storage.type=elasticsearch"}
+	flag.Parse()
+	handler, err := NewSpanHandlerBuilder(
+		builder.Options.LoggerOption(zap.NewNop()),
+		builder.Options.ElasticSearchOption(&escfg.Configuration{
+			Servers: []string{"127.0.0.1"},
+		}),
+	)
+	assert.NoError(t, err)
+	assert.NotNil(t, handler)
+}
+
+func TestNewSpanHandlerBuilderElasticSearchFailure(t *testing.T) {
+	originalArgs := os.Args
+	defer func() {
+		os.Args = originalArgs
+	}()
+	os.Args = []string{"test", "--span-storage.type=elasticsearch"}
+	flag.Parse()
+	handler, err := NewSpanHandlerBuilder()
+	assert.Error(t, err)
```

Comment:
this returns a specific error, right? Let's check for it explicitly with ErrorEquals, not just for an abstract error presence.